packages/material-ui/src/TextareaAutosize/TextareaAutosize.js

```diff
@@ -115,17 +123,25 @@ const TextareaAutosize = React.forwardRef(function TextareaAutosize(props, ref)
     });
   }, [maxRows, minRows, props.placeholder]);
 
+  const resizeObserverRef = React.useRef(null);
+
   React.useEffect(() => {
     const handleResize = debounce(() => {
       renders.current = 0;
       syncHeight();
     });
 
-    const containerWindow = ownerWindow(inputRef.current);
-    containerWindow.addEventListener('resize', handleResize);
+    try {
+      resizeObserverRef.current = new ResizeObserver(handleResize);
+    } catch (err) {
+      resizeObserverRef.current = MockResizeObserver(); // Prevent crash for old browsers
+    }
```

Comment:
No need to try catch nor to create a MockResizeObserve. We can test if ResizeObserver is defined and no nothing if its not.

Reply:
This also prevents test failure, so I think we still need to include it?